Library/Formula/iojs.rb

```diff
@@ -0,0 +1,125 @@
+    system "make", "install"
+    (bin+"iojs").write iojs_script
+
+    resource("npm").stage buildpath/"npm_install"
+
+    # make sure npm can find iojs
+    ENV.prepend_path "PATH", libexec/"iojs/bin"
+    # set log level temporarily for npm's `make install`
+    ENV["NPM_CONFIG_LOGLEVEL"] = "verbose"
+
+    cd buildpath/"npm_install" do
+      system "./configure", "--prefix=#{libexec}/npm"
+      system "make", "install"
+    end
+    (bin+"npm-iojs").write npm_script
+
+    if build.with? "completion"
+      bash_completion.install \
+        buildpath/"npm_install/lib/utils/completion.sh" => "npm"
+    end
+  end
+
+  # This tells iojs where to find npm modules, and executes the hidden iojs.
+  def iojs_script
+    <<-EOS.undent
+      #!/usr/bin/env bash
+      export NODE_PATH=#{HOMEBREW_PREFIX}/share/iojs/
+      exec #{libexec}/iojs/bin/iojs "$@"
+    EOS
+  end
+
+  # This tells npm where to find iojs.
+  def npm_script
+    <<-EOS.undent
+      #!/usr/bin/env bash
+      export PATH=#{libexec}/iojs/bin:$PATH
+      exec #{HOMEBREW_PREFIX}/share/iojs/bin/npm "$@"
+    EOS
+  end
+
+  def post_install
+    iojs_bindir = HOMEBREW_PREFIX/"share/iojs/bin"
+    iojs_bindir.mkpath
+    iojs_modules = HOMEBREW_PREFIX/"share/iojs/lib/node_modules"
+    iojs_modules.mkpath
+    npm_exec = iojs_modules/"npm/bin/npm-cli.js"
+    # Kill npm but preserve all other modules across iojs updates/upgrades.
+    rm_rf iojs_modules/"npm"
+
+    cp_r libexec/"npm/lib/node_modules/npm", iojs_modules
+    ln_sf npm_exec, iojs_bindir/"npm"
+
+    # Let's do the manpage dance. It's just a jump to the left.
+    # And then a step to the right, with your hand on rm_f.
+    ["man1", "man3", "man5", "man7"].each do |man|
+      mkdir_p HOMEBREW_PREFIX/"share/man/#{man}"
+      rm_f Dir[HOMEBREW_PREFIX/"share/man/#{man}/{npm.,npm-,npmrc.}*"]
+      Dir[libexec/"npm/share/man/#{man}/npm*"].each { |f| ln_sf f, HOMEBREW_PREFIX/"share/man/#{man}" }
+    end
+
+    npm_root = iojs_modules/"npm"
+    npmrc = npm_root/"npmrc"
+    npmrc.atomic_write("prefix = #{HOMEBREW_PREFIX}/share/iojs")
+  end
+
+  def caveats; <<-EOS.undent
+    If you update npm itself, do NOT use the npm update command.
+    The upstream-recommended way to update npm is:
+      npm-iojs install -g npm@latest
+
+    Executing `npm-iojs` in your terminal client will automatically,
+    temporarily shunt iojs's Node symlink to the front of your $PATH.
+    This allows npm to find the desired version of Node rather than
+    any other version of Node available along the $PATH, and consequently
+    you can directly install modules with `npm-iojs install -g grunt`.
+
+    However, your modules installed via `npm` do not currently reside
+    in your `$PATH` as there's a limited amount Homebrew can do to isolate
+    those. If you wish to use those modules you should permanently
+    prepend your `$PATH` like such:
+      export PATH=#{HOMEBREW_PREFIX}/share/iojs/bin:#{libexec}/iojs/bin:$PATH
```

Comment:
On this line: If you follow that line, you can just do ` npm install -g blah ` or similar. We could simplify this, but I wanted to give maintainers and people options for discussion for now, and I wanted to give developers a way to activate the iojs npm without having it absolutely conflict with the ` Node ` npm. This reasonably-hideous workaround achieves that.